### ISSUE TITLE
i need LORA support

### ISSUE THREAD

**Issue by sev7eny** [comfyanonymous/ComfyUI_TensorRT#42]
please add lora ande controlnet support!!!

**Comment by gseth** [comment]
LoRA support is cheesed in. When building the engine, connect it with a Lora. Now, the engine model will have the Lora baked in. In the final workflow, you won't be able to change the Lora Strength, though.

**Comment by BigPeng** [comment]
> LoRA support is cheesed in. When building the engine, connect it with a Lora. Now, the engine model will have the Lora baked in. In the final workflow, you won't be able to change the Lora Strength, though.

it works, but the result is worse

**Comment by BigPeng** [comment]
> > LoRA support is cheesed in. When building the engine, connect it with a Lora. Now, the engine model will have the Lora baked in. In the final workflow, you won't be able to change the Lora Strength, though.
> 
> it works, but the result is worse

I found that tensorrt only supports 77 tokens. If it exceeds the limit, an error will be reported. The exceeded prompt part is invalid, resulting in inconsistent picture results.
Error Code 3: API Usage Error (Parameter check failed, condition: engineDims.d[i] == dims.d[i]. Static dimension mismatch while setting input shape for context. Set dimensions are [1,154,2048]. Expected dimensions are [1,77,2048].)

**Comment by comfyanonymous** [comment]
>Is there any way to support more than 77 tokens?

Yes, use the node to build a dynamic engine and increase context_max to more than 1. The max amount of tokens is context_max * 77

**Comment by catboxanon** [comment]
FYI, on-the-fly LoRA support for TensorRT has been available in the web UI since October 2023. https://github.com/NVIDIA/Stable-Diffusion-WebUI-TensorRT

**Comment by phazei** [comment]
So if the webUI supports lora's, what's stopping support for that here?

**Comment by flynet** [comment]
@phazei Lora works with TensorRT by taking a model, attaching the required number of Loras with the desired strength, and then converting it to TensorRT. As far as I understand, TensorRT significantly speeds up performance in workflows that are already “tuned.” It shouldn’t be used on raw workflows.

This issue is waiting for support of controlnet not lora
